test/text/test_render_styled.py

```diff
@@ -87,3 +90,189 @@ def test_render_styled_newpos(tmp_path):
             doc.line(doc.x, doc.y - 3, doc.x, doc.y + 3)
 
     assert_pdf_equal(doc, HERE / "render_styled_newpos.pdf", tmp_path)
+
+
+def test_cell_newpos(tmp_path):
+    """
+    Verify that cell() places the new position
+    in the right places in all possible combinations of alignment,
+    new_x, and new_y.
+
+    Note:
+        cell() doesn't process align="J", and uses "L" instead.
+    """
+    doc = fpdf.FPDF()
+    doc.set_font("helvetica", style="U", size=24)
+    doc.set_margin(10)
+    twidth = 100
+
+    for i, item in enumerate(data):
+        i = i % 4
+        if i == 0:
+            doc.add_page()
+        doc.x = 20
+        doc.y = 20 + (i * 20)
+        s = item[0]
+        align = item[1]
+        newx = item[2]
+        newy = item[3]
+        doc.cell(
+            twidth,
+            txt=s,
+            border=1,
+            align=align,
+            new_x=newx,
+            new_y=newy,
+        )
+        # mark the new position in the file with crosshairs for verification
+        with doc.rotation(i * -15, doc.x, doc.y):
+            doc.circle(doc.x - 3, doc.y - 3, 6)
+            doc.line(doc.x - 3, doc.y, doc.x + 3, doc.y)
+            doc.line(doc.x, doc.y - 3, doc.x, doc.y + 3)
+
+    assert_pdf_equal(doc, HERE / "cell_newpos.pdf", tmp_path)
+
+
+def test_multi_cell_newpos(tmp_path):
+    """
+    Verify that multi_cell() places the new position
+    in the right places in all possible combinations of alignment,
+    new_x, and new_y.
+
+    Note:
+        multi_cell() doesn't use align="J" on the first line, and
+        uses "L" instead. new_x is relative to the last line.
+    """
+    doc = fpdf.FPDF()
+    doc.set_font("helvetica", style="U", size=24)
+    doc.set_margin(10)
+    twidth = 100
+
+    for i, item in enumerate(data):
+        i = i % 4
+        if i == 0:
+            doc.add_page()
+        doc.x = 20
+        doc.y = 20 + (i * 20)
+        s = item[0]
+        align = item[1]
+        newx = item[2]
+        newy = item[3]
+        doc.multi_cell(
+            twidth,
+            txt=s + "\n-",
+            border=1,
+            align=align,
+            new_x=newx,
+            new_y=newy,
+        )
+        # mark the new position in the file with crosshairs for verification
+        with doc.rotation(i * -15, doc.x, doc.y):
+            doc.circle(doc.x - 3, doc.y - 3, 6)
+            doc.line(doc.x - 3, doc.y, doc.x + 3, doc.y)
+            doc.line(doc.x, doc.y - 3, doc.x, doc.y + 3)
+
+    assert_pdf_equal(doc, HERE / "multi_cell_newpos.pdf", tmp_path)
+
+
+data2 = (
```

Comment:
Could you find more explicit names for those constants? 😅

Reply:
already done due to the notice above... :wink: